### PR TITLE
feat(secrets-engine): implement and test AWS secrets manager service

### PR DESCRIPTION
**Pull request by nik-ewf**

### CODE REVIEW

File: libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+  KEYS = 'keys',
+  RSA_KEY = 'rsa_key',
+}
+
+@Injectable()
+export class AwsSecretsManagerService
+  extends SecretsEngineService
+  implements OnModuleInit
+{
+  private readonly logger = new Logger(VaultService.name);
```
Comment (hejkerooo) [inline libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:39]:
```suggestion
  private readonly logger = new Logger(AwsSecretsManagerService.name);
```

File: libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+  SecretsEngineService,
+  SetCertificateDetailsResponse,
+  SetEncryptionKeysResponse,
+  SetPrivateKeyResponse,
+  SetRSAPrivateKeyResponse,
+} from '../secrets-engine.interface';
+import { VaultService } from './vault.service';
+import { Span } from 'nestjs-otel';
+
+enum PATHS {
```
Comment (kim-energyweb) [inline libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:25]: Isn't it better to have this paths shared between vaults?

Comment (kim-energyweb) [reply libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:25]: What do you think @hejkerooo ?

Comment (nik-ewf) [reply libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:25]: Yep you're right, I think I meant to do that but forgot 🤦

File: libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+    ) as PromiseRejectedResult[];
+    if (errors.length > 0) {
+      this.logger.error(errors.map(({ reason }) => reason.message).join(', '));
+    }
+
+    const [privateKey, certificate, caCertificate] = responses;
+
+    return {
+      privateKey:
+        privateKey.status === 'fulfilled'
```
Comment (vikaskumr) [inline libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:239]: Just a thought. Can we put these statuses in a const file.

Comment (nik-ewf) [reply libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:239]: It's the status returned by `Promise.allFulfilled`, either `rejected` or `fulfilled`. Don't think we really need them unless we are using `allFullfilled` a lot?

File: libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+
+  @Span('aws_ssm_getEncryptionKeys')
+  async getEncryptionKeys(): Promise<EncryptionKeys | null> {
+    const command = new GetSecretValueCommand({
+      SecretId: `${this.prefix}${PATHS.KEYS}`,
+    });
+
+    return this.client
+      .send(command)
+      .then(({ SecretString }) => JSON.parse(SecretString))
```
[the remark this answers is not in the document]
Comment (nik-ewf) [reply libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:339]: Ah yeah, it's a 65kb limit. I'll split it out into different secrets like `setCertificateDetails`

File: libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts [hunk 5]
```diff
@@ -0,0 +1,345 @@
...
+    });
+
+    return this.client
+      .send(command)
+      .then((response) => {
+        this.logger.log(response);
+        return response;
+      })
+      .catch(async (err) => {
+        if (err instanceof ResourceNotFoundException) {
```
[the remark this answers is not in the document]
Comment (nik-ewf) [reply libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:67]: Yep I'll split it out into another function

File: libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (nik-ewf) [reply libs/dsb-client-gateway-secrets-engine/src/lib/service/aws-secrets-manager.service.ts:339]: Removed get/setEncryptionKeys implementations after slack convo

**Review by hejkerooo — APPROVED** [review]

**Review by kim-energyweb — APPROVED** [review]
LGTM

**Pull request merged**